server/src/plugins/image-migration.ts

```diff
@@ -0,0 +1,31 @@
+import { getEvents, updateEvent } from '../db';
+import fetch from 'node-fetch';
+import dotenv from 'dotenv';
+
+import { uploadFile } from './google-storage-utils';
+
+dotenv.config();
+
+export async function migrateEventImagesToGoogle(){
+  const events = await getEvents();
+  for await (let event of events) {
+    if(event.image_url){
+      console.log(event.image_url)
+      const response = await fetch(event.image_url)
+      const file = await response.buffer();
+
+      const filename = event.fancy_id + '-' + event.id + '-logo.png'
+      const google_image_url = await uploadFile(filename, 'image/png', file);
+      if (google_image_url) {
+        console.log('--->' + google_image_url);
+        await updateEvent(event.fancy_id, event.event_host_id, {
+          event_url: event.event_url,
+          image_url: google_image_url
+        });
+      }
+    }
+  }
+
+}
```

Comment:
We can filter all the events that doesn't have the `image_url` key so we avoid nesting conditionals. Also we can use destructuring and template strings to interpolate variables.

```typescript
export async function migrateEventImagesToGoogle() {
  const events = await getEvents();
  const eventsWithImages = events.filter(event => event.hasOwnProperty('image_url'));

  for await (let event of eventsWithImages) {
    const {
      id,
      fancy_id: fancyId,
      event_url: eventUrl,
      image_url: imageUrl,
      event_host_id: eventHostId,
    } = event;

    const response = await fetch(imageUrl);
    const file = await response.buffer();

    const filename = `${fancyId}-${id}-logo.png`;
    const googleImageUrl = await uploadFile(filename, 'image/png', file);

    if (googleImageUrl) {
      await updateEvent(fancyId, eventHostId, { event_url: eventUrl, image_url: googleImageUrl });
    }
  }
}

```

Reply:
`const eventsWithImages = events.filter(event => event.hasOwnProperty('image_url'));` returns the event if the image is an empty string, any workarround to keep it in one line?

Reply:
Oh that could happen! Since what the filter needs back it's a boolean and that's what we are returning it we can append another condition using `AND` operator.

Option 1:
```typescript
const eventsWithImages = events.filter(event => event.hasOwnProperty('image_url') && event.image_url !== '');
```
Explanation: now we can access `.image_url` because we checked if existed first.

Option 2:
```typescript
const eventsWithImages = events.filter(event => event.hasOwnProperty('image_url') && event.image_url);
```
Explanation: Since the empty string `''` it's a falsy value in JavaScript we can avoid comparing it against an empty string.

Choose the one you find more you think is readable to you and your peers.